packages/mailer/src/GenericMailer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Tempest\Mail;
+
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Tempest\Mail\Exceptions\MailerTransportWasMissing;
+use Tempest\Mail\MailerConfig;
+use Tempest\View\ViewRenderer;
+
+use function Tempest\map;
+
+/**
+ * Generic mailer based on Symfony transports.
+ */
+final class GenericMailer implements Mailer
+{
+    public function __construct(
+        private MailerConfig $mailerConfig,
+        private ViewRenderer $viewRenderer,
+        private ?TransportInterface $transport = null,
+    ) {
+        $this->transport ??= $this->createTransport();
+    }
+
+    public function send(Email $email): SentEmail
+    {
+        $symfonyEmail = map($email)
+            ->with(fn (Email $from) => new EmailToSymfonyEmailMapper($this->mailerConfig, $this->viewRenderer)->map($from, null))
+            ->do();
+
+        return new SentGenericEmail(
```

Comment:
Wouldn't it make more sense to dispatch an `EmailSent` event instead of returning a new value object? That way we can hook into the event from any place.

Reply:
This might be a good idea, but that changes the testing implementation a bit... I'll explore that

Reply:
I explored it and that makes testing too hard. But I did add event support 👍